sdlib/d/gc/arena.d

```diff
@@ -269,6 +272,41 @@ private:
 		(cast(Arena*) &this).shrinkAllocImpl(e, index, pages, delta);
 	}
 
+	bool growAlloc(shared(ExtentMap)* emap, Extent* e, uint pages) shared {
+		assert(!e.isHuge(), "Does not support huge!");
+		assert(pages > 0 && pages > e.pageCount, "Invalid page count!");
+
+		auto n = e.hpdIndex;
+
+		if (n + pages > PagesInHugePage) {
+			return false;
+		}
+
+		uint delta = pages - e.pageCount;
+		uint index = n + e.pageCount;
```

Comment:
You should avoid fetching the same field on multiple occasions. This may seems trivial to optimize away, but it is not so. If there is any write in between the loads, the compiler has to prove that the write doesn't alias, and it's actually quite difficult and often fails.